format/view-spec.md

```diff
@@ -42,12 +42,28 @@ An atomic swap of one view metadata file for another provides the basis for maki
 
 Writers create view metadata files optimistically, assuming that the current metadata location will not be changed before the writer's commit. Once a writer has created an update, it commits by swapping the view's metadata file pointer from the base location to the new location.
 
+### Materialized Views
+
+Materialized views are a type of view with precomputed results from the view query stored as a table.
+When queried, engines may return the precomputed data for the materialized views, shifting the cost of query execution to the precomputation step.
+
+Iceberg materialized views are implemented as a combination of an Iceberg view and an underlying Iceberg table, known as the storage table, which stores the precomputed data.
+The metadata for a materialized view extends the Iceberg view metadata, adding a pointer to the precomputed data and refresh information to determine if the data is still fresh. 
+The refresh information is composed of data about the so-called "source tables", which are the tables referenced in the query definition of the materialized view. 
+The storage table can be in the states of "fresh", "stale" or "invalid", which are determined from the following situations:
+* **fresh** -- The `snapshot_id`s of the last refresh operation match the current `snapshot_id`s of the source tables.
+* **stale** -- The `snapshot_id`s do not match, indicating that a refresh operation needs to be performed to capture the latest source table changes.
+* **invalid** -- The current `version_id` of the materialized view does not match the `refresh-version-id` of the refresh state. 
```

Comment:
```suggestion
* **fresh** -- The `snapshot_id`s of the last refresh operation match the current `snapshot_id`s of all the source tables.
* **stale** -- The `snapshot_id`s do not match for at-least one source table, indicating that a refresh operation needs to be performed to capture the latest source table changes.
* **invalid** -- The current `version_id` of the materialized view does not match the `refresh-version-id` of the refresh state. 
```

nitpicking but I think it helps to be explicit since I don't see much discussion about how would multi-table MVs work yet. Or is it not considered in scope for the initial spec changes?

Reply:
If you are talking about different physical representations of the same view, we did discuss this and decided it was out of scope.

Reply:
@hashhar I think your changes make it more precise. Thank you!

Reply:
@bennychow I was not thinking of multi-engine/multiple materializations. I was thinking of a MV which selects from multiple source tables like: `CREATE MATERIALIZED VIEW mv AS SELECT t1.key, t2.value FROM t1 INNER JOIN t2 ON t1.id=t2.id`

Reply:
oic.. yes, this spec supports your example with t1 and t2... or if they were views.. and even if there was time travel on the tables.